pkg/extension/sumologicextension/extension.go

```diff
@@ -39,6 +38,7 @@ type SumologicExtension struct {
 	baseUrl          string
 	conf             *Config
 	logger           *zap.Logger
+	creds            credsGetter
```

Comment:
```suggestion
	creds            CredsGetter
```

This should be an interface so that we can customize the extension with the behavior of different credentials getters + I'd call this (the field) credentialsGetter or so because otherwise I'd be inclined to assign obtained credentials in `Start` to this field insteaf of `registrationInfo` on line 103